### PR TITLE
Feature: Auth0 Frontend integration

### PR DESCRIPTION
**Pull request by Umang-Rajkarnikar**
User can signup, login, and logout of application using auth0

Added authentication guard to protect /satellite-passes and /user-info pages

Restructured client repository to use /pages folder structure instead of app

https://github.com/RishiVaya/Lower_Earth_Orbiters/assets/59628796/3ee3fd8e-8ce8-4217-8b3f-4d44e7904026

### CODE REVIEW

File: src/leo-client-app/src/components/FuturePasses.tsx [hunk 1]
```diff
@@ -41,29 +43,35 @@ const FuturePasses: React.FC = () => {
...
-                {passPair[1].type === 'Exit' && (
-                  <>
-                    {passPair[1].time}
-                  </>
-                )}
+              <td style={{ border: "1px solid #000", padding: "8px" }}>
+                {passPair[1].type === "Exit" && <>{passPair[1].time}</>}
               </td>
             </tr>
           ))}
```
Comment (quinnha) [inline src/leo-client-app/src/components/FuturePasses.tsx:77]: can you add a screenshot of what this styling is? i know it doesnt matter but hard to approve without seeing what it looks like

File: src/leo-client-app/src/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  description: "Generated by create next app",
+};
+
+export default function MyApp({ Component, pageProps }: AppProps) {
+  return (
+    <UserProvider>
+      <Component {...pageProps} />
+    </UserProvider>
+  );
+}
```
Comment (quinnha) [inline src/leo-client-app/src/pages/_app.tsx:20]: what does this file do?

File: src/leo-client-app/src/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+// ----------------------------------------------------------------------
+
+export default function Index() {
+  const { pathname, replace, prefetch } = useRouter();
+
+  useEffect(() => {
+    // if (pathname === PATH_DASHBOARD.projects.root) {
```
Comment (quinnha) [inline src/leo-client-app/src/pages/index.tsx:10]: remove comment

File: src/leo-client-app/src/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    replace("/satellite-passes");
+    // }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (quinnha) [inline src/leo-client-app/src/pages/index.tsx:13]: also remove comment

File: src/leo-client-app/src/pages/_app.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import type { AppProps } from "next/app";
+
+const inter = Inter({ subsets: ["latin"] });
+
+export const metadata: Metadata = {
+  title: "Create Next App",
+  description: "Generated by create next app",
+};
+
+export default function MyApp({ Component, pageProps }: AppProps) {
```
Comment (quinnha) [inline src/leo-client-app/src/pages/_app.tsx:14]: could be nice to add a small comment over what this does

File: src/leo-client-app/src/pages/satellite-passes/index.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+"use client";
+
+import Image from "next/image";
+// import styles from "./page.module.css";
```
Comment (quinnha) [inline src/leo-client-app/src/pages/satellite-passes/index.tsx:4]: remove

File: src/leo-client-app/src/pages/satellite-passes/index.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import SatelliteInfo from "@/components/SatelliteInfo";
+import FuturePasses from "@/components/FuturePasses";
+import { withPageAuthRequired } from "@auth0/nextjs-auth0/client";
+
+function SatelliteInfoPage() {
+  return (
+    <main>
+      <a href="/api/auth/logout">Logout</a>
+
```
Comment (quinnha) [inline src/leo-client-app/src/pages/satellite-passes/index.tsx:13]: nit: extra line

File: src/leo-client-app/src/pages/user-info/index.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import Image from "next/image";
+// import styles from "./page.module.css";
```
Comment (quinnha) [inline src/leo-client-app/src/pages/user-info/index.tsx:4]: remove comment

File: src/leo-client-app/src/pages/user-info/index.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    // };
+    // axios
+    //   .request(options)
+    //   .then(function (response) {
+    //     console.log(response.data);
+    //   })
+    //   .catch(function (error) {
+    //     console.error(error);
+    //   });
+  };
```
Comment (quinnha) [inline src/leo-client-app/src/pages/user-info/index.tsx:33]: if not needed, remove

**Review by quinnha — CHANGES_REQUESTED** [review]
add some comments for clarity, and remove code that is commented out

also can you attach a screenshot of what your changes look like in your PR so we know what it is? can be a vid of the user flow or just ui

File: src/leo-client-app/src/components/SatelliteInfo.tsx [hunk 10]
```diff
@@ -44,13 +45,13 @@ const SatelliteInfo: React.FC = () => {
   });
 
   const fetchData = () => {
-    fetch('http://localhost:3001/getSatelliteInfo') // TODO: Fix endpoint for when server is hosted
+    fetch("http://localhost:3001/getSatelliteInfo") // TODO: Fix endpoint for when server is hosted
```
Comment (quinnha) [inline src/leo-client-app/src/components/SatelliteInfo.tsx:48]: u funny for this change

File: src/leo-client-app/src/components/SatelliteInfo.tsx [hunk 11]
```diff
@@ -66,42 +67,53 @@ const SatelliteInfo: React.FC = () => {
     };
   }, []);
 
-  const { positionEci, velocityEci, longitude, latitude, height, azimuth, elevation, rangeSat } = state;
+  const {
```
Comment (quinnha) [inline src/leo-client-app/src/components/SatelliteInfo.tsx:70]: nit: extra space

File: src/leo-client-app/src/components/SatelliteInfo.tsx (hunk 10 above)
Comment (Umang-Rajkarnikar) [reply src/leo-client-app/src/components/SatelliteInfo.tsx:48]: lol i think that's my prettier acting up

File: src/leo-client-app/src/pages/_app.tsx [hunk 12]
```diff
@@ -11,6 +11,10 @@ export const metadata: Metadata = {
   description: "Generated by create next app",
 };
 
+/*
```
Comment (quinnha) [inline src/leo-client-app/src/pages/_app.tsx:14]: nit: follow more conventional docstring format:

i.e
/*
* Component is the active page
* UserProvider is a wrapper that allows us to access
* auth0 features globally throughout the app
/*

**Review by quinnha — COMMENTED** [review]
why is the font times new roman instead of whatever we had default 💀 

can you also change the table so its more in line with the black background instead of the white? just so the poc looks slightly better

other than that looks solid

File: src/leo-client-app/src/pages/_app.tsx (hunk 12 above)
Comment (quinnha) [reply src/leo-client-app/src/pages/_app.tsx:14]: ok the formatting looks weird but just add a * before every line

File: src/leo-client-app/src/components/SatelliteInfo.tsx (hunk 11 above)
Comment (Umang-Rajkarnikar) [reply src/leo-client-app/src/components/SatelliteInfo.tsx:70]: that's how eslint destructures an object if the line is too long

Comment (quinnha) [reply src/leo-client-app/src/components/SatelliteInfo.tsx:70]: man thats ugly (i should check what linter im using)

File: src/leo-client-app/src/pages/_app.tsx (hunk 12 above)
Comment (quinnha) [reply src/leo-client-app/src/pages/_app.tsx:14]: if this is a eslint issue then ignore

**Comment by quinnha** [comment]
lgtm, waiting on @RishiVaya to review

**Pull request merged**